exporter/containerimage/attestations.go

```diff
@@ -1,10 +1,184 @@
 package containerimage
 
 import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/fs"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	"github.com/moby/buildkit/cache"
+	"github.com/moby/buildkit/exporter/attestation"
+	gatewaypb "github.com/moby/buildkit/frontend/gateway/pb"
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/solver"
+	"github.com/moby/buildkit/solver/result"
+	"github.com/moby/buildkit/version"
 	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/spdx/tools-golang/jsonloader"
+	"github.com/spdx/tools-golang/jsonsaver"
+	"github.com/spdx/tools-golang/spdx"
 )
 
 var intotoPlatform ocispecs.Platform = ocispecs.Platform{
 	Architecture: "unknown",
 	OS:           "unknown",
 }
+
+// supplementSBOM modifies SPDX attestations to include the file layers
+func supplementSBOM(ctx context.Context, s session.Group, target cache.ImmutableRef, targetRemote *solver.Remote, refs map[string]cache.ImmutableRef, att result.Attestation) (result.Attestation, error) {
```

Comment:
this would be better with 2 functions in series. `supplementSBOM()` should just take `[]byte` as an input.